specs/WebMessageObjects.md

```diff
@@ -157,11 +234,165 @@ interface ICoreWebView2WebMessageReceivedEventArgs2 : ICoreWebView2WebMessageRec
+  /// model of WebView2](https://learn.microsoft.com/microsoft-edge/webview2/concepts/process-model).
+  /// Note that the only validation done in this function is that the path is a valid
+  /// full path and otherwise the function will succeed.
+  /// Any other state validation will be done when this handle is accessed from web content
+  /// and will cause DOM exceptions if access operations fail.
+  HRESULT CreateWebFileSystemHandle(
+      [in] LPCWSTR path,
+      [in] COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION permission,
+      [in] COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND kind,
+      [out, retval] ICoreWebView2FileSystemHandle** fileSystemHandle);
+
+  /// Create a ICoreWebView2File from a file path. The object created is a
+  /// representation of a DOM [File](https://developer.mozilla.org/docs/Web/API/File)
+  /// object.
+  /// The `path` is the full path pointed to the file. It must be a valid file 
+  /// that the app has read access for and is available to read or creation will fail
+  /// with E_INVALIDARG.
+  /// (See Footnote 1)
+  HRESULT CreateWebFile(
+      [in] LPCWSTR path,
+      [out, retval] ICoreWebView2File** file);
+
+  /// Create a generic object collection. 
+  HRESULT CreateObjectCollection(
+      [in] UINT32 length,
+      [in, size_is(length)] IUnknown** items,
+      [out, retval] ICoreWebView2ObjectCollection** objectCollection);
+}
+
+[uuid(5220d097-c583-4115-8e08-1af2100338c2), object, pointer_default(unique)]
+interface ICoreWebView2_24 : ICoreWebView2_23 {
+  /// Same as PostWebMessageAsJson, but also has support for posting DOM objects
+  /// to page content. The `additionalObjects` property in the DOM MessageEvent
+  /// fired on the page content is a ArrayLike list of
+  /// DOM objects that can be posted to the web content. Currently these can be
+  /// the following types and `null`:
+  /// | Win32             | DOM type    |
+  /// |-------------------|-------------|
+  /// | ICoreWebView2File | [File](https://developer.mozilla.org/docs/Web/API/File) |
+  /// | ICoreWebView2FileSystemHandle | [FileSystemHandle](https://developer.mozilla.org/docs/Web/API/FileSystemHandle) |
+  /// | nullptr           | null        |
+  /// The objects are posted the to web content following the structured-clone
+  /// semantics, meaning only objects that can be cloned can be posted.
+  /// They will also behave as in they are created on the web content they are
+  /// posted to. For example, if a FileSystemFileHandle is posted to a web content
+  /// it can only be transferred via postMessage to other web content with the same origin.
+  HRESULT PostWebMessageAsJsonWithAdditionalObjects([in] LPCWSTR webMessageAsJson, [in] ICoreWebView2ObjectCollectionView* additionalObjects);
+}
 ```
 
 ```c#
 namespace Microsoft.Web.WebView2.Core
 {
+    enum CoreWebView2FileSystemHandlePermission
+    {
+        /// Read-only permission for FileSystemHandle
+        CoreWebView2FileSystemHandlePermissionReadonly = 0,

```

Comment:
In WinRT, the values of an enum do not need to be prefixed with the name of the enum. i.e. this could be 
```
enum CoreWebView2FileSystemHandlePermission
{
    Readonly,
    Readwrite
}
```

Similarly for CoreWebView2FileSystemHandleKind.

Reply:
And `Readonly` and `Readwrite` should be camel case:
```
ReadOnly
ReadWrite
```

Reply:
Indeed, the earlier C# sample used the shorter names with the correct capitalization.

(PS: PascalCase is first-letter-of-each-word. camelCase is lowercase first letter, because camels have a hump in the middle.)